jdaviz/core/data_formats.py

```diff
@@ -112,3 +122,145 @@ def identify_data(filename, current=None):
+
+
+def _get_wcs(filename, header):
+    """
+    Get gwcs.wcs.WCS or astropy.wcs.WCS from FITS file.
+    """
+    try:
+        with asdf_in_fits.open(filename) as af:
+            wcs = af.tree['meta']['wcs']
+
+    # if the file doesn't have ASDF-in-FITS, then
+    # the 'meta' key doesn't exist, yielding a KeyError:
+    except KeyError:
+        # fall back on using astropy WCS:
+        wcs = WCS(header)
+
+    return wcs
+
+
+def identify_helper(filename, ext=1):
+    """
+    Guess the appropriate viz helper for a data file.
+
+    Parameters
+    ----------
+    filename : str (path-like)
+        Name for a local data file.
+    ext : int
+        Extension from the FITS file.
+
+    Returns
+    -------
+    helper_name : str
+        Name of the best-guess helper for ``filename``.
+    """
+    supported_dtypes = [
+        Spectrum1D,
+        SpectrumList,
+        SpectrumCollection,
+        CCDData
+    ]
+
+    if filename.lower().endswith('asdf'):
+        # ASDF files are only supported in jdaviz for
+        # Roman WFI 2D images, so suggest imviz:
+        return 'imviz'
+
+    header = fits.getheader(filename, ext=ext)
+    data = fits.getdata(filename, ext=ext)
+    wcs = _get_wcs(filename, header)
+    has_spectral_axis = 'spectral' in wcs.world_axis_object_classes
+
+    n_axes = (
+        int(has_spectral_axis) +
+
+        sum([component[0] in ['celestial', 'angle']
+             for component in wcs.world_axis_object_components]) -
+
+        # remove any slit_frame axis from the count
+        (0 if not hasattr(wcs, 'available_frames') else
+         int('slit_frame' in wcs.available_frames))
+    )
+
+    # use astropy to recognize some data formats:
+    possible_formats = {}
+    for cls in supported_dtypes:
+        fmt = registry.identify_format(
+            'read', cls, filename, None, {}, {}
+        )
+        if fmt:
+            possible_formats[cls] = fmt
+
+    # If CCDData is the only match:
+    if len(possible_formats) == 1:
+        only_key, only_value = possible_formats.popitem()
+        if only_key == CCDData:
+            # could be 2D spectrum or 2D image. break tie with WCS:
+            if has_spectral_axis:
+                if n_axes > 1:
+                    return 'specviz2d'
+                return 'specviz'
+            elif not isinstance(data, fits.BinTableHDU):
+                return 'imviz'
+
+    # Ensure specviz is chosen when ``data`` is a table or recarray
+    # and there's a "known" spectral column name:
+    if isinstance(data, (fits.BinTableHDU, fits.fitsrec.FITS_rec)):
+        # now catch spectra in FITS tables, looking for
+        # columns with "wave" or "flux" in the names:
+        table_columns = [getattr(col, 'name', col).lower() for col in data.columns]
+
+        # these are "known" prefixes for column names
+        # in FITS tables of spectral observations
+        known_spectral_columns = [
+            'wave',
+            'flux'
+        ]
```

Comment:
I worry about hard-coding these kinds of column names, but can't think of a more general way to do this.  I suggested we could try to auto-identify wavelength or flux columns based on any attached units on the column.  Maybe that's out of scope for this PR though.

Reply:
> out of scope for this PR though.

I think so. Let's get v0.1 done and iterate.